api/opds.py

```diff
@@ -278,7 +295,12 @@ def acquisition_links(
                 # set. There is one link for the delivery mechanism
                 # that was locked in, and links for any streaming
                 # delivery mechanisms.
+                #
+                # Since the delivery mechanism has already been locked in,
+                # we choose not to use visible_delivery_mechanisms --
+                # they already chose it and they're stuck with it.
                 for lpdm in active_license_pool.delivery_mechanisms:
+
                     if lpdm is active_loan.fulfillment or lpdm.delivery_mechanism.is_streaming:
```

Comment:
There's a very small inconsistency here: if one of the streaming content types is hidden, it will still show up at this point, because we're not iterating over `visible_delivery_mechanisms`. I don't think it's worth fixing this problem because hopefully we can remove this entire feature once we fix Open eBooks.

Reply:
But this is only for streaming content types and not other content types?

Reply:
Yeah, the logic is: show a link if it's a streaming content type or if it's the content type that the patron "locked in" when they borrowed the book.

If the "locked in" content type is hidden, we want to show it anyway, because the patron is already committed to it. Hiding it will just make the situation worse.

If one of the streaming types is hidden, it would be better to hide it here as well -- presumably we hid it for a reason, and we don't want it showing up at all.